### PR TITLE
Create OLM Bundle

### PR DESCRIPTION
**Pull request by bcrochet**
This creates the bundle necessary for integrating with Operator Lifecycle
Manager (OLM).

Signed-off-by: Brad P. Crochet <brad@redhat.com>

**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -177,6 +187,33 @@ vendor: $(GO)
 tools: $(GO)
 	./hack/install-tools.sh
 
+# Generate bundle manifests and metadata, then validate generated files.
+bundle: gen-crds manifests
```
Comment (phoracek) [inline Makefile:191]: The upgrade guide for 1.2.0 says this should use kustomize: https://github.com/operator-framework/operator-sdk/blob/master/website/content/en/docs/upgrading-sdk-version/v1.2.0.md#golang-based-operators-update-makefiles-bundle-target

**Review by phoracek — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (bcrochet) [reply Makefile:191]: That is planned. Just want to get to this sooner than later.

File: api/v1beta1/nmstate_types.go [hunk 2]
```diff
@@ -30,15 +31,22 @@ type NMStateSpec struct {
...
 // +kubebuilder:storageversion
+// +kubebuilder:subresource=status
 
 // NMState is the Schema for the nmstates API
 type NMState struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              NMStateSpec `json:"spec,omitempty"`
+	Spec              NMStateSpec   `json:"spec,omitempty"`
+	Status            NMStateStatus `json:"status,omitempty"`
```
Comment (qinqon) [inline api/v1beta1/nmstate_types.go:49]: I don't see the controller code that update the status.

**Review by qinqon — CHANGES_REQUESTED** [review]
Looks good overall: a nit and a question.

File: api/v1beta1/nmstate_types.go (hunk 2 above)
Comment (bcrochet) [reply api/v1beta1/nmstate_types.go:49]: The controller doesn't do anything with that. OLM uses the Conditions for it's own operator.

**Pull request merged**